### ISSUE TITLE
downsample issue: before analysis_functions is applied

### ISSUE THREAD

**Issue by JingjinW** [DynaSim/DynaSim#573]
We already know that using downsampling would miss out information. But it should be pointed out that downsample is actually performed before analysis_functions.

analysis_functions is applied on downsampled data rather than unscaled data:
This causes issue especially if you: 1. didn't monitor spikes; 2. do anything about spikes with your function. Because your function will get your spikes from voltage, which is already downsampled.
Basically I thought the order was: get the raw data, analyze **unscaled** data, downsample it, save.
But in reality it seems to be: get the raw data, downsample it, analyze **scaled** data, save data.

**Comment by JingjinW** [comment]
This could cause subtle issue. If I have spikes monitored, spikes will mostly be preserved  (discount really close up doublets). Anything that uses spikes data will be ok. But if I don't, spikes will be derived from voltage, which is downsampled, and the spikes will be much sparse than it actually is, without any warning. Anything based on that would be inaccurate.

**Comment by kupiqu** [comment]
Yes, at least a warning would be nice.

Could you please paste minimal examples of the two scenarios you're mentioning here?
